lib/code_corps/github/adapters/comment.ex

```diff
@@ -40,11 +39,53 @@ defmodule CodeCorps.GitHub.Adapters.Comment do
   ]
 
   @doc ~S"""
-  Adapts the GitHub payload into attributes for creating or updating a
-  `CodeCorps.GithubComment` record.
+  Converts a GitHub Issue Comment payload into a set of attributes suitable for
+  creating or updating a `CodeCorps.GithubComment`
   """
   @spec to_github_comment(map) :: map
   def to_github_comment(%{} = payload) do
     payload |> MapTransformer.transform(@github_comment_mapping)
   end
+
+  @autogenerated_github_keys ~w(created_at id updated_at)
+
+  @doc ~S"""
+  Converts a `CodeCorps.Comment` into a set of attributes suitable to create or
+  update an GitHub Issue Comment through the GitHub API.
+  """
+  @spec to_api(Comment.t) :: map
+  def to_api(%Comment{} = comment) do
+    comment
+    |> Map.from_struct
+    |> MapTransformer.transform_inverse(@mapping)
+    |> Map.drop(@autogenerated_github_keys)
+    |> add_user_info_header(comment)
+  end
```

Comment:
This makes perfect sense to me, and I would agree.

One thing to also consider is that the payload for an issue from an event is different from the payload for an issue from the endpoint itself. This might get us into trouble in the future.